types/react-codemirror/index.d.ts

```diff
@@ -1,22 +1,41 @@
-// Type definitions for React Codemirror v0.2.6
+// Type definitions for React Codemirror v1.0.0
 // Project: https://github.com/JedWatson/react-codemirror
 // Definitions by: Vicky Lai <https://github.com/velveret>
+//                 Rudi Chen <https://github.com/rudi-c>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 2.4
 
 /// <reference types="react"/>
 /// <reference types="codemirror"/>
 
 declare namespace ReactCodeMirror {
     interface ReactCodeMirrorProps extends React.Props<ReactCodeMirror> {
-        onChange?: (newValue: string) => any; // called when a change is made
-        onFocusChange?: (focused: boolean) => any; // called when the editor is focused or loses focus
-        onScroll?: (scrollInfo: CodeMirror.ScrollInfo) => any; // called when the editor is scrolled
-        options?: CodeMirror.EditorConfiguration; // options passed to the CodeMirror instance
-        path?: string; // the identifying name for the textarea
-        value?: string; // the editor value
-        className?: string; // CSS className for the outer element
-        codeMirrorInstance?: CodeMirror.Editor; // the CodeMirror instance
+        /** Automatically focuses the editor when it is mounted (default false) */
+        autoFocus?: boolean;
+        /** Automatically persist changes to underlying textarea (default false) */
+        autoSave?: boolean;
```

Comment:
this prop is listed in the docs but isn't actually in the library - see https://github.com/JedWatson/react-codemirror/pull/22#issuecomment-320414389. I'm fine with leaving this in for now to match the react-codemirror docs and so we don't have to update again once it's restored.